src/ucp/core/ucp_worker.c

```diff
@@ -2959,13 +2966,28 @@ ucp_worker_do_keepalive_progress(ucp_worker_h worker)
             goto out;
         }
 
+#if UCS_ENABLE_ASSERT
+        ucp_ep_ext_control(ep)->ka_count++;
+        /* Difference between the number of EP keepalive rounds and total number
+         * of KA rounds done on Worker must be <= 2, because keepalive could be
+         * done twice for the same EP in case of number of EPs was decreased */
+        ucs_assertv((ucp_ep_ext_control(ep)->ka_count /
+                             (worker->keepalive.round_count + 1)) <= 2,
+                    "ep %p: keepalive.round_count=%" PRIu64 " ep.ka_count=%"
+                    PRIu64, ep, worker->keepalive.round_count,
+                    ucp_ep_ext_control(ep)->ka_count);
+#endif
```

Comment:
maybe move this to ucp_ep_do_keepalive?

Reply:
we should do after we know that there is no lanes returned NO_RESOURCE

Reply:
we should be able to know it in ucp_ep_do_keepalive, given it's the function which updates worker->keepalive.lane_map

Reply:
yes, we able to know, but then it will be checked twice